h/form.py

```diff
@@ -58,14 +65,25 @@ def __call__(self, template_name, **kwargs):
             template_name += '.jinja2'
 
         template = self._get_env().get_template(template_name)
+        context = self._context.copy()
+        context.update(kwargs)
 
-        return jinja2.Markup(template.render(**kwargs))
+        return jinja2.Markup(template.render(context))
 
 
-def initialize_deform_renderer(config):
-    deform.Form.set_default_renderer(
-        Jinja2Renderer(config.get_jinja2_environment()))
+def form_factory(request, config):
+    """
+    Return a callable which creates deform.Form instances for a request.
+
+    The callable accepts the same arguments as deform.Form but the resulting
+    form will be configured to have access to properties of the request in its
+    templates, such as feature flags.
+    """
+    renderer = Jinja2Renderer(config.get_jinja2_environment(), {
+        'feature': request.feature })
+    return partial(deform.Form, renderer=renderer)
```

Comment:
So this now creates an instance of `Jinja2Renderer` each time you create a form. This isn't necessarily the end of the world, but it's also not necessary.

Perhaps save a single instance of the renderer on the registry and then just deal with the context in the `render` function -- i.e. the `__call__` method.